File: frontend/lib/api/aogRecovery.ts
// Typed REAL-mode client for the backend AOG recovery synthesis endpoint
// (backend/app/api/v1/aog.py: GET /aog-events/recovery-status/{aircraft_id}).
// organization_id is never sent — the backend derives it from the JWT.

import { apiRequest } from "@/lib/apiClient";

export interface BackendRecoveryBlocker {
  category: string;
  description: string;
  record_type: string;
  record_id: string | null;
  who_should_act: string;
  dependency: string;
}

export interface BackendAogRecoveryStatus {
  aircraft_id: string;
  registration: string;
  is_aog: boolean;
  aog_event_id: string | null;
  aog_status: string | null;
  severity: string | null;
  work_order_id: string | null;
  release_readiness_status: string | null;
  tat_status: string | null;
  tat_reason: string | null;
  blockers: BackendRecoveryBlocker[];
  next_best_action: BackendRecoveryBlocker | null;
  technician_authorization: string;
  eta: string;
  compliance_status: string;
  data_completeness: string;
}

export const aogRecoveryApi = {
  getRecoveryStatus: (accessToken: string, aircraftId: string) =>
    apiRequest<BackendAogRecoveryStatus>(`/aog-events/recovery-status/${aircraftId}`, {
      accessToken,
    }),
};

File: backend/app/schemas/aog_recovery.py
from pydantic import BaseModel, ConfigDict


class RecoveryBlockerResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    category: str
    description: str
    record_type: str
    record_id: str | None
    who_should_act: str
    dependency: str


class CriticalPathStageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    stage: str
    status: str
    reason: str
    record_type: str | None
    record_id: str | None
    next_action: str | None


class AogRecoveryStatusResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    aircraft_id: str
    registration: str
    is_aog: bool
    aog_event_id: str | None
    aog_status: str | None
    severity: str | None
    work_order_id: str | None
    release_readiness_status: str | None
    tat_status: str | None
    tat_reason: str | None
    blockers: list[RecoveryBlockerResponse]
    next_best_action: RecoveryBlockerResponse | None
    critical_path: list[CriticalPathStageResponse]
    technician_authorization: str
    eta: str
    compliance_status: str
    data_completeness: str
